examples/source_examples/plaid/connector.py

```diff
@@ -0,0 +1,98 @@
+# You need to provide your credentials for this example to work.
+# See the Technical Reference documentation (https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update)
+# and the Best Practices documentation (https://fivetran.com/docs/connectors/connector-sdk/best-practices) for details
+import json
+
+from plaid import ApiClient, Configuration, Environment
+from plaid.api import plaid_api
+from plaid.model.accounts_get_request import AccountsGetRequest
+from fivetran_connector_sdk import Connector, Operations as op
+
+
+def initialize_plaid_client(configuration):
+    plaid_configuration = Configuration(
+        host=Environment.Production,
+        api_key={
+            "clientId": configuration["client_id"],
+            "secret": configuration["client_secret"],
+        },
+    )
+    plaid_client = ApiClient(plaid_configuration)
+    return plaid_api.PlaidApi(plaid_client)
+
+
+# Define the schema function, which lets you configure the schema your connector delivers.
+# See the technical reference documentation for more details on the schema function:
+# https://fivetran.com/docs/connectors/connector-sdk/technical-reference#schema
+# The schema function takes one parameter:
+# - configuration: a dictionary that holds the configuration settings for the connector.
+def schema(configuration: dict):
+    return [
+        {
+            "table": "accounts",  # Name of the table in the destination.
+            "primary_key": ["account_id"],  # Primary key column(s) for the table.
+            "columns": {  # Define the columns and their data types.
+                "account_id": "STRING",  # String column for the account_id.
+                "holder_category": "STRING",  # String column for the holder_category.
+                "mask": "STRING",  # String column for the mask.
+                "name": "STRING",  # String column for the name.
+                "official_name": "STRING",  # String column for the official_name.
+                "persistent_account_id": "STRING",  # String column for the persistent_account_id.
+                "subtype": "STRING",  # String column for the subtype.
+                "type": "STRING",  # String column for the type.
+            },
+        }
+    ]
+
+
+# Define the update function, which is a required function, and is called by Fivetran during each sync.
+# See the technical reference documentation for more details on the update function
+# https://fivetran.com/docs/connectors/connector-sdk/technical-reference#update
+# The function takes two parameters:
+# - configuration: a dictionary that contains any secrets or payloads you configure when deploying the connector
+# - state: a dictionary that contains whatever state you have chosen to checkpoint during the prior sync.
+# The state dictionary is empty for the first sync or for any full re-sync.
+def update(configuration: dict, state: dict):
```

Comment:
please break this into small methods for better readable code, so it can easily extended by our customers.